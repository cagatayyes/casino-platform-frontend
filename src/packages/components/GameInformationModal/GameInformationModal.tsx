
import { ModalPlacement } from 'casino/components/Modal/Modal.type';
import { Modal } from 'casino/components/Modal/Modal';
import { ButtonVariant, IButtonProps } from 'casino/components/Button/Button.types';
import GameInformation from 'casino/components/GameInformation/GameInformation';
import { getGameNameFromLink } from 'casino/utils';

interface IGameInformationModal {
    gameId: number;
	selectedGameLink: string;
    isShowModal:boolean;
    showgameinfomodal: (gameId:number, gameLink: string)=>void;
	maxWidth:string;
	handlePlayNow: (gameLink: string, gameName: string, gamelaunchpoint: string, categoryName?: string) => void;
}

export const GameInformationModal = (
	{
		gameId,
		selectedGameLink,
		isShowModal,
		showgameinfomodal,
		maxWidth,
		handlePlayNow
	}: IGameInformationModal): JSX.Element => {

	const playGameNow = (): void => {
		const gameName = getGameNameFromLink(selectedGameLink);

		showgameinfomodal(gameId, selectedGameLink);
		handlePlayNow(selectedGameLink, gameName, 'Information Modal');
	};

	const GameInfoModalActions: IButtonProps[] = [
		{
			label: 'PLAY NOW',
			className: 'flex w-full mx-1 mt-[14px] !w-[180px] !h-[38px] items-center',
			onClick: playGameNow,
			color: ButtonVariant.Yellow,
			variant: ButtonVariant.Yellow,
			noStyle: false,
			labelStyle:'text-[16px] font-bold tracking-widest leading-[14px] text-[14px] tracking-[16%] h-[14px]'
		}
	];

	return(
		<Modal
			show={isShowModal && gameId > -1}
			placement={ModalPlacement.CENTER}
			actions={GameInfoModalActions}
			overideContainerStyles='bg-[var(--main-bg-color)] px-0 py-0 h-[429px] w-full max-w-screen-sm mx-[24px]'
			hasCloseButton={true}
			onClose={(): void => showgameinfomodal(gameId, selectedGameLink)}
		>
			{/* TODO: The complete information is not provided by provider */}
			<div className='h-[350px]'>
				<GameInformation
					gameId={gameId}
					overideImageStyles="absolute top-0 left-1/2 -translate-x-1/2 -translate-y-1/2"
					overideImageContainerStyles="h-[90px]"
					maxWidth={maxWidth}
				/>
			</div>
		</Modal>
	);
};
