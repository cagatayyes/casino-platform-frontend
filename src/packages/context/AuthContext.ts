import { createContext } from 'react';

// eslint-disable-next-line @typescript-eslint/naming-convention
interface IAuthContext {
    authToken: string;
    setAuthToken: (authToken: string) => void;
}

export const AuthContext = createContext<IAuthContext>({
	authToken: '',
	setAuthToken: () => { }
});
