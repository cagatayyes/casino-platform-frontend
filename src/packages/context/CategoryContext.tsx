import { CATEGORIES_ENDPOINT } from 'casino/constants';
import { useAPI } from 'casino/hooks';
import { ICategory, ICategoryResponse } from 'casino/types';
import React, { createContext } from 'react';

export const CategoryContext = createContext<ICategoryResponse | undefined>(undefined);

export const CategoryProvider: React.FC<{ children: React.ReactNode }> = ({ children }) => {
	const { fetchData: fetchCategoryData, data: categoryList, isLoading: isCategoryLoading, error: categoryFetchError } =
		useAPI<ICategory[]>({ endpoint: CATEGORIES_ENDPOINT });

	return (
		<CategoryContext.Provider value={{ categoryList, isCategoryLoading, categoryFetchError, fetchCategoryData }}>
			{children}
		</CategoryContext.Provider>
	);
};

