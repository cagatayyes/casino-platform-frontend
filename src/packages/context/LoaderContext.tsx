import { Loader } from 'casino/components';
import { createContext, useState } from 'react';

export interface ILoaderContext {
	isLoading: boolean;
	showLoader: (loaderMessage: string) => void;
	hideLoader: () => void;
}

export const LoaderContext = createContext<ILoaderContext | undefined>(undefined);

export const LoaderProvider = ({ children }: {children: React.ReactNode }): JSX.Element => {
	const [isLoading, setIsLoading] = useState(false);
	const [loadingText, setLoadingText] = useState('');

	const showLoader = (loaderMessage: string): void => {
		setIsLoading(true);
		setLoadingText(loaderMessage);
	};
	const hideLoader = (): void => setIsLoading(false);

	return (
		<LoaderContext.Provider value={{ isLoading, showLoader, hideLoader }}>
			<>
				{isLoading && <Loader loaderText={loadingText} />}
				{children}
			</>
		</LoaderContext.Provider>
	);
};
