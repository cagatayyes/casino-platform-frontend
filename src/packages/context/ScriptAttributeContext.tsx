import { CASINO_CLIENT_LOGO, CASINO_NOT_FOUND, CASINO_WIDGET_SCRIPT_ID } from 'casino/constants';
import { ICasinoScriptAttribute, IScriptAttribute, SCRIPT_ATTRIBUTE_KEYS } from 'casino/types';
import React, { Dispatch, createContext, useEffect, useReducer } from 'react';

export const scriptAttributeInitialState: IScriptAttribute = {
	token: '',
	currency: 'NGN',
	maxwidth: '640px',
	timezone: 'Africa/Lagos',
	clientlogo: CASINO_CLIENT_LOGO,
	pagenotfoundlogo: CASINO_NOT_FOUND,
	isTokenChanged: false,
	baseURL: '/casino'
};

type Action = | { type: 'SET_SCRIPT_ATTRIBUTE', payload: IScriptAttribute }
	| { type: 'SET_TOKEN_CHANGED', payload: boolean };

const scriptAttributeReducer = (state: IScriptAttribute, action: Action): IScriptAttribute => {
	switch (action.type) {
	case 'SET_SCRIPT_ATTRIBUTE': {
		const isTokenChanged = action.payload.token !== state.token;

		return { ...state, ...action.payload, isTokenChanged };
	}
	case 'SET_TOKEN_CHANGED': {
		return { ...state, isTokenChanged: action.payload };
	}
	default:
		return state;
	}
};

export interface IScriptAttributeContext {
    scriptAttributes: IScriptAttribute,
	scriptAttributesDispatch: Dispatch<Action>;
}

const getScriptAttributes = (): IScriptAttribute => {
	const casinoScriptAttributes: NamedNodeMap | undefined = document.getElementById(CASINO_WIDGET_SCRIPT_ID)?.attributes;

	return Array.from(casinoScriptAttributes || []).reduce((attribute: ICasinoScriptAttribute, { name, value }) => {
		if (SCRIPT_ATTRIBUTE_KEYS.includes(name)) {
			attribute[name] = value;
		}

		return attribute;
	}, { ...scriptAttributeInitialState });
};

const observeScriptAttributes = (scriptAttributesDispatch: Dispatch<Action>): (() => void) => {
	const casinoWidgetScript = document.getElementById(CASINO_WIDGET_SCRIPT_ID);
	if (!casinoWidgetScript) return () => {};

	const config = { attributes: true };
	const mutationObserverCallback = (mutationsList: MutationRecord[]): void => {
		for (const mutation of mutationsList) {
			if (mutation.type === 'attributes' && mutation.attributeName) {
				const name: string = mutation.attributeName;
				const newValue: string | null = casinoWidgetScript.getAttribute(name);

				if (SCRIPT_ATTRIBUTE_KEYS.includes(name)) {
					scriptAttributesDispatch({
						type: 'SET_SCRIPT_ATTRIBUTE',
						payload: { [name]: newValue } as ICasinoScriptAttribute,
					});
				}
			}
		}
	};

	const observer = new MutationObserver(mutationObserverCallback);
	observer.observe(casinoWidgetScript, config);

	return (): void => observer.disconnect();
};

export const ScriptAttributeContext = createContext<IScriptAttributeContext | undefined>(undefined);

export const ScriptAttributeProvider: React.FC<{ children: React.ReactNode }> = ({ children }) => {
	const initialState = getScriptAttributes();
	const [scriptAttributes, scriptAttributesDispatch] = useReducer(scriptAttributeReducer, initialState);

	useEffect(() => {
		const casinoWidgetObserver =  observeScriptAttributes(scriptAttributesDispatch);

		return () => {
			casinoWidgetObserver();
		};
	}, []);

	return (
		<ScriptAttributeContext.Provider value={{ scriptAttributes, scriptAttributesDispatch }}>
			{children}
		</ScriptAttributeContext.Provider>
	);
};
