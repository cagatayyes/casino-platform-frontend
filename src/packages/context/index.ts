export * from './LoaderContext';
export * from './AuthContext';
export * from './ScriptAttributeContext';
export * from './CategoryContext';
