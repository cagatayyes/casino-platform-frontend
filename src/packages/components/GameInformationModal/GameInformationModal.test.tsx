/* eslint-disable react/prop-types */
import { render } from '@testing-library/react';
import { GameInformationModal } from './GameInformationModal';
import React from 'react';
import { ScriptAttributeProvider } from '../../context/ScriptAttributeContext';

jest.mock('swiper/react', () => ({
	Swiper: ({ children }): JSX.Element => <div>{children}</div>,
	SwiperSlide: ({ children }): JSX.Element => <div>{children}</div>,
}));

jest.mock('swiper/modules', () => ({
	Swiper: ({ children }): JSX.Element => <div>{children}</div>,
	SwiperSlide: ({ children }): JSX.Element => <div>{children}</div>,
}));

describe('Game Information Modal', () => {
	it('should render Game information modal', () => {
		const component = render(
			<ScriptAttributeProvider>
				<GameInformationModal
					gameId={14}
					selectedGameLink=''
					isShowModal={true}
					showGameInfoModal={jest.fn()}
					maxWidth='640px'
					handlePlayNow={jest.fn()}
				/>
			</ScriptAttributeProvider>
		);
		expect(component).toMatchSnapshot();
	});
});
