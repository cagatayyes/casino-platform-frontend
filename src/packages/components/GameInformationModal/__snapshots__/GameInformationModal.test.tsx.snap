// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game Information Modal should render Game information modal 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style="overflow: hidden;"
  >
    <div />
    <div
      class=""
      id="casino-modal-container"
    >
      <div
        class="absolute flex justify-center w-full z-2222 h-screen items-center"
        placement="center"
        style="z-index: 2222; top: 0px;"
      >
        <div
          class="absolute backdrop bg-[#000000] left-0 opacity-50 top-1000 w-full h-[100%]"
        />
        <div
          class="relative modal p-[15px] rounded bg-[var(--main-bg-color)] px-0 py-0 h-[429px] w-full max-w-screen-sm mx-[24px]"
          placement="center"
        >
          <div
            class=""
            data-testid="casino-modal-container"
            placement="center"
          >
            <span
              class="material-symbols-outlined absolute top-[5px] right-[5px] text-[#fff] cursor-pointer"
              style="font-size: 20px;"
            >
              close
            </span>
            <div
              class=""
              data-testid="casino-modal-content"
              placement="center"
            >
              <div
                class="h-[350px]"
              >
                <div
                  class="flex flex-col w-full h-fit max-w-screen-sm bg-[var(--main-bg-color)]"
                >
                  <div
                    class="flex items-center h-screen justify-center"
                    style="max-width: 640px;"
                  >
                    <div
                      class="flex relative h-[250px] w-full justify-center mt-[-100px]"
                    >
                      <img
                        alt="App Loader"
                        class="animate-pulse h-24 w-24"
                        height="24"
                        src="https://casino-cdn.leetent.co.uk/images/loader-1.svg"
                        width="24"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="flex justify-center"
            >
              <button
                class="border-[1px] text-sm tracking-[-0.2px] h-[40px] rounded-[4px] font-bold flex justify-center items-center transition-colors flex w-full mx-1 mt-[14px] !w-[180px] !h-[38px] items-center px-[18px] border-[#F4BB40] !bg-[#F4BB40]
				text-[#010C2B]"
              >
                <span
                  class="flex text-[16px] font-bold tracking-widest leading-[14px] text-[14px] tracking-[16%] h-[14px]"
                >
                  PLAY NOW
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
